src/app/filelogger.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2016  sledgehammer999 <hammered999@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include <QDateTime>
+#include <QDir>
+#include <QFile>
+#include <QTextStream>
+#include "filelogger.h"
+#include "base/logger.h"
+#include "base/utils/fs.h"
+
+FileLogger::FileLogger(const QString &path, const bool backup, const int maxSize, const bool deleteOld, const int age, const FileLogAgeType ageType)
+    : m_backup(backup)
+    , m_maxSize(maxSize)
+    , m_logFile(nullptr)
+{
+    m_flusher.setInterval(0);
+    m_flusher.setSingleShot(true);
+    connect(&m_flusher, SIGNAL(timeout()), SLOT(flushLog()));
+
+    changePath(path);
+    if (deleteOld)
+        this->deleteOld(age, ageType);
+
+    const Logger* const logger = Logger::instance();
+    foreach (const Log::Msg& msg, logger->getMessages())
+        addLogMessage(msg);
+
+    connect(logger, SIGNAL(newLogMessage(const Log::Msg &)), SLOT(addLogMessage(const Log::Msg &)));
+}
+
+FileLogger::~FileLogger()
+{
+    if (!m_logFile) return;
+    closeLogFile();
+    delete m_logFile;
+}
+
+void FileLogger::changePath(const QString& newPath)
+{
+    QString tmpPath = Utils::Fs::fromNativePath(newPath);
```

Comment:
I am not sure what you guys are talking about. There is an agreed rule to use internally paths with "/" instead of "\" even if the called functions understand both(like QDir). Or are you talking about something else entirely?